components/nexus-core/src/main/java/org/sonatype/nexus/internal/orient/SharedHazelcastPlugin.java

```diff
@@ -27,11 +31,25 @@
 public class SharedHazelcastPlugin
     extends OHazelcastPlugin
 {
-  private static final Logger log = LoggerFactory.getLogger(SharedHazelcastPlugin.class);
+  public static final String INSTANCE_NAME = "nexus";
 
   @Override
   protected HazelcastInstance configureHazelcast() {
-    // HACK: Orient-components are not proper Sisu-components, but share the provider lookup
-    return new HazelcastInstanceProvider().get();
+    return (HazelcastInstance) Proxy.newProxyInstance(HazelcastInstance.class.getClassLoader(),
+        new Class<?>[] { HazelcastInstance.class }, new InvocationHandler()
+        {
+          private Object delegate = checkNotNull(Hazelcast.getHazelcastInstanceByName(INSTANCE_NAME));
+
+          public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+            if ("shutdown".equals(method.getName())) {
+              delegate = null; // detach proxy, but keep delegate alive
+              return null;
+            }
+            if (delegate == null) {
+              throw new HazelcastInstanceNotActiveException();
+            }
+            return method.invoke(delegate, args);
+          }
+        });
   }
```

Comment:
btw I think we may eventually need to create separate bundles for distributed components, and provide different implementations of DatabaseServer based on if the fabric stuff is enabled or not.